diff --git a/CMakeLists.txt b/CMakeLists.txt
index b0eabe7..94db283 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -109,10 +109,12 @@ endif()
 if(WIN32)
     list(REMOVE_ITEM au_sources "${CMAKE_SOURCE_DIR}/src/logger_ansi.cc")
     list(REMOVE_ITEM au_sources "${CMAKE_SOURCE_DIR}/src/logger_dummy.cc")
-else(CYGWIN OR UNIX)
+endif()
+if(CYGWIN OR UNIX)
     list(REMOVE_ITEM au_sources "${CMAKE_SOURCE_DIR}/src/logger_win.cc")
     list(REMOVE_ITEM au_sources "${CMAKE_SOURCE_DIR}/src/logger_dummy.cc")
-else()
+endif()
+if(APPLE)
     list(REMOVE_ITEM au_sources "${CMAKE_SOURCE_DIR}/src/logger_win.cc")
     list(REMOVE_ITEM au_sources "${CMAKE_SOURCE_DIR}/src/logger_ansi.cc")
 endif()
